workspaces/global-floating-action-button/plugins/global-floating-action-button/src/components/FloatingButton.tsx

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// how many sub-menu items do we support ?
+
+import * as React from 'react';
+import classnames from 'classnames';
+import Fab from '@mui/material/Fab';
+import { makeStyles, Typography } from '@material-ui/core';
+import { useTheme } from '@mui/material/styles';
+import CloseIcon from '@mui/icons-material/Close';
+import Tooltip from '@mui/material/Tooltip';
+import Slide from '@mui/material/Slide';
+
+type FloatingButtonAction = {
+  // The floating button action will override these properties when there's only one action.
+  icon: React.ReactElement;
+  size?: 'small' | 'medium' | 'large' | string;
+  position?: { top?: string; left?: string; right?: string; bottom?: string };
+  hide?: boolean;
+  color?:
+    | 'default'
+    | 'error'
+    | 'info'
+    | 'inherit'
+    | 'primary'
+    | 'secondary'
+    | 'success'
+    | 'warning'
+    | string;
+  onClick?: React.MouseEventHandler;
+  url?: string;
+  label?: string | React.ReactNode;
+  toolTip?: string;
+  actions?: FloatingButtonAction[];
+};
+
+const useStyles = makeStyles(() => ({
+  button: {
+    zIndex: 200,
+    display: 'flex',
+    position: 'fixed',
+    maxWidth: '150px',
+    gap: '10px',
+  },
+}));
+
+const fab = (
+  icon: React.ReactElement,
+  size: 'small' | 'medium' | 'large' | string,
+  color?: any,
+  label?: string | React.ReactNode,
+  onClick?: any,
+  url?: string,
+  tooltip?: string,
+) => {
```

Comment:
Feels like a component, but isn't one. Couldn't yo convert this to a component where a FAB (your FAB type) is a prop?